test/src/test/java/org/corfudb/runtime/LogReplicationLogicalGroupClientShimTest.java

```diff
@@ -0,0 +1,281 @@
+package org.corfudb.runtime;
+
+import org.corfudb.runtime.LogReplication.ClientDestinationInfoKey;
+import org.corfudb.runtime.LogReplication.ClientRegistrationId;
+import org.corfudb.runtime.LogReplication.ClientRegistrationInfo;
+import org.corfudb.runtime.LogReplication.DestinationInfoVal;
+import org.corfudb.runtime.collections.CorfuStoreShim;
+import org.corfudb.runtime.collections.Table;
+import org.corfudb.runtime.collections.TableOptions;
+import org.corfudb.runtime.view.AbstractViewTest;
+import org.corfudb.test.SampleSchema.ManagedResources;
+import org.corfudb.test.SampleSchema.Uuid;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+public class LogReplicationLogicalGroupClientShimTest extends AbstractViewTest {
+    private static final String namespace = "CorfuSystem";
+    private static final String registrationTableName = "LogReplicationRegistrationTable";
+    private static final String metadataTableName = "LogReplicationModelMetadataTable";
+    private static final String clientName = "client";
+
+    private static Table<ClientRegistrationId, ClientRegistrationInfo, ManagedResources> replicationRegistrationTable;
+    private static Table<ClientDestinationInfoKey, DestinationInfoVal, ManagedResources> sourceMetadataTable;
+    private static CorfuRuntime runtime;
+    private static LogReplicationLogicalGroupClient client;
+
+    private CorfuRuntime getTestRuntime() {
+        return getDefaultRuntime();
+    }
+
+    @Before
+    public void loadProperties() throws Exception {
+        runtime = getTestRuntime();
+        CorfuStoreShim store = new CorfuStoreShim(runtime);
```

Comment:
We shouldnt use CorfuStoreShim here because your code uses the real CorfuStore for the db operations.

CorfuStoreShim is an abstraction over CorfuStore and was specifically written for external verticals.  We do not use it inside LR.